### DISCUSSION TITLE
Chat agent that can be primed with a prompt

### DISCUSSION THREAD

**Discussion by naciriy2** [FlowiseAI/Flowise#867] [Q&A]
Hi! What's the right flowchart and variables to set up a chat agent that can be primed with a specific prompt and has memory? I am new at this so please if you'd be kind to help.  The Bot answers the initial questions according to the prompt but not the follow-on questions. It keeps going back to the original question.
![Chartflow](https://github.com/FlowiseAI/Flowise/assets/143841804/86ea9937-5d6f-40c7-8ce5-99af48b4b49f)
![System message](https://github.com/FlowiseAI/Flowise/assets/143841804/df49444c-4ed1-4e15-bec5-05508800bcd0)
![Prompt values](https://github.com/FlowiseAI/Flowise/assets/143841804/05fca9e9-fd31-4e70-8058-19cb176a53cc)

**Comment by cocobeach** [comment]
The way this could work
<img width="1511" alt="Screenshot 2023-09-13 at 15 15 54" src="https://github.com/FlowiseAI/Flowise/assets/128893/3a1efc35-f907-4163-8d66-c84decedfb74">
In the community flows you have the "conversational Agent" you will notice you need a tool calculator, and you also have the ability to browse, this should do the trick for what you ask, but you can add a vector retriever tool, and a pinecone DB for more personalised context.
